### ISSUE TITLE
Error: Duplicate identifier 'LibraryManagedAttributes'

### ISSUE THREAD

**Issue by Spenhouet** [facebook/react#13686]
I suddenly couldn't build my react project. This error seems to appear for many people.

- https://stackoverflow.com/questions/52339225/react-typescript-2312-14-duplicate-identifier-librarymanagedattributes
- https://stackoverflow.com/questions/52323056/typescript-error-duplicate-identifier-librarymanagedattributes
- https://github.com/ant-design/ant-design/issues/11697
- https://stackoverflow.com/questions/52399839/typescript-duplicate-identifier-librarymanagedattributes

`yarn build`:

```
Failed to compile.

C:/dev/project/frontend/node_modules/@types/prop-types/node_modules/@types/react/index.d.ts
(2312,14): Duplicate identifier 'LibraryManagedAttributes'.

error Command failed with exit code 1.
```

It looks like `LibraryManagedAttributes` is defined in the following files:

 - `node_modules/@types/react/index.d.ts`
 - `node_modules/@types/prop-types/node_modules/@types/react/index.d.ts`
 - `node_modules/@types/react-overlays/node_modules/@types/react/index.d.ts`
 - ....

**What is the expected behavior?**

No error.

**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**

```
"@types/react": "^16.4.14",
"@types/react-dom": "^16.0.7",
"@types/prop-types": "^15.5.5",
"react": "^16.5.2",
"react-dom": "^16.5.2",
"prop-types": "^15.6.2",
```

**Comment by gaearon** [comment]
React does not have any relationship with `node_modules/@types`. This sounds like TypeScript definitions. We don’t maintain TypeScript definitions for React — DefinitelyTyped community does. So perhaps that would be the best place to report this. It’s not actionable for us because we don’t own these files.

**Comment by ekilah** [comment]
@Spenhouet  and anyone else getting here via google: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/33822#issuecomment-515390815

tl;dr:

```
yarn add yarn-deduplicate --dev
yarn yarn-deduplicate yarn.lock
```

^ this got me out of types hell and lots of manual `yarn.lock` modifications, YMMV of course
